ark/utils/metacluster_remap_gui/metaclustergui.py

```diff
@@ -0,0 +1,463 @@
+import warnings
+
+import numpy as np
+import ipywidgets as widgets
+import matplotlib.pyplot as plt
+import matplotlib.patheffects as path_effects
+import seaborn as sns
+from scipy.cluster.hierarchy import dendrogram
+from scipy.stats import zscore
+from IPython.display import display
+from scipy.stats.stats import F_onewayBadInputSizesWarning
+
+from .zscore_norm import ZScoreNormalize
+from .colormap_helper import distinct_cmap
+from .metaclusterdata import MetaClusterData
+from .throttle import throttle
+
+# Third party ipympl causing this in it's backend_agg startup code
+warnings.filterwarnings("ignore", message="nbagg.transparent is deprecated")
+
+DEBUG_VIEW = widgets.Output(layout={'border': '1px solid black'})
+
+
+class MetaClusterGui():
+    def __init__(self, metaclusterdata, heatmapcolors=sns.diverging_palette(240, 10, n=3, as_cmap=True), width=17.0, debug=False, enable_throttle=True):  # noqa
```

Comment:
This can be indented instead to avoid the linter errors right?

Reply:
Of course, I always try both ways and make a judgement call on readability. One technique to mitigate masking lint error is disabling only one specific lint rather than noqa catchall, pylint supports this for example, but pycodestyle doesn't seem to.

Reply:
You didn't like the result of formatting it something like this?
``` 
def __init__ (xxxx, 
            yyy, zzz)
```

Reply:
### tldr
I can and will do it this way, I see it is used throughout the rest of the codebase, it's not horrible.

### long

1. Bias against for wrapping any line which starts a block because it (slightly) degrades the block structure and negatively affects "skimmability"
2. Favor wrapping with constant indent of 4 to minimize noise (nearby lines touched) when commit edits. Otherwise renaming the in the first line forces you to adjust multiple lines. It also give you more space when the first line is already long.
    ```
    myobject.my_list_real_long_variable_naming = [
        a,
        b,
        c,
    ]
    ```
    vs
    ```
    myobject.my_list_real_long_variable_naming = [
                                                   a,
                                                   b,
                                                   c,
                                                     ]
    ```

source: https://www.youtube.com/watch?v=x-kB2o8sd5c ~35 minutes in.

Reply:
Totally, sounds good